### DISCUSSION TITLE
[Meta] Labels for enhancement categories

### DISCUSSION THREAD

**Discussion by DNin01** [ScratchAddons/ScratchAddons#5793] [Discussion]
It can sometimes be useful to see what kind of enhancement certain issues are making.

We can add labels to help separate issues into categories of changes:

- `category: accessibility` (Addresses an accessibility issue)
- `category: feature` (A new feature or improvement to a feature)
- `category: ux` (An improvement to the user experience)
etc.

This discussion will be used to talk about how we should go about adding these category labels, if this is something we're all interested in. It might even help us better prioritize issues that need attention based on what kinds of changes they make.

**Comment by Hans5958** [comment]
Why not doing it on `scope:`?

**Reply by DNin01** [reply]
Scope seems to me like it tells us where the change will be, whether it's to an addon, a webpage, or a server. That's why I wanted to make a separate group of labels for this.

**Reply by Hans5958** [reply]
I would say that when doing this, I generalize the definition of "scope" much to also cover general categories/groupings such as `scope: tech debt`, `scope: test failure`, etc (there's also `scope: from turbowarp` for some reason). It **may** a good idea to differentiate between the two.

**Reply by DNin01** [reply]
Hmm... okay.

So, you mean it might be okay to use scope in labels like `scope: accessibility` too?

**Reply by mxmou** [reply]
We already have `scope: design`, so I think it would make sense to add `scope: accessibility`.

**Reply by DNin01** [reply]
> We already have `scope: design`, so I think it would make sense to add `scope: accessibility`.

Alright, then! I've gone ahead and did that.

**Reply by WorldLanguages** [reply]
Just created a `scope: performance` label too.

**Reply by DNin01** [reply]
I went ahead and made a `scope: user experience` label as well.

**Reply by WorldLanguages** [reply]
@DNin01 Nice one. I have thought of that one multiple times but never actually created it.

**Comment by DNin01** [comment]
A `type: checklist` label could also be helpful, maybe?

**Reply by Hans5958** [reply]
A more usual word would be a "megathread". But that works too.

**Comment by DNin01** [comment]
I think `status: needs discussion` might be useful too. We could, for example, use it in PRs without issues.

**Reply by Samq64** [reply]
why not? There are already a bunch of PRs that are stalled because of that, even ones with linked issues.

**Reply by DNin01** [reply]
Done!

**Comment by DNin01** [comment]
Would having a label for string changes be a good idea?

How would we describe strings to someone who doesn't know, by the way?

**Reply by WorldLanguages** [reply]
No idea

**Comment by DNin01** [comment]
Some sort of "response to update" label might help us separate issues that have to do with Scratch updates, too.

**Reply by WorldLanguages** [reply]
Yeah, but good luck finding a way to name this label 🤔

**Reply by Hans5958** [reply]
You can do `scope: upstream`. Upstream here refers to something related to what we depend, such as Scratch itself, browser bugs that we can't do anything (looking at you, Apple), and something related to our libraries. This includes anything regarding to Manifest v3.

**Reply by DNin01** [reply]
Done!

**Comment by DNin01** [comment]
What do you think of a slightly brightened color for the https://github.com/ScratchAddons/ScratchAddons/labels/scope%3A%20addon and https://github.com/ScratchAddons/ScratchAddons/labels/scope%3A%20addon%2Ejson labels?
`#f26ad0` → `#fd76d8`?

Might help with text contrast in light mode?

**Reply by DNin01** [reply]
I've finally went ahead and updated it.

**Reply by DNin01** [reply]
And maybe we should change the color of https://github.com/ScratchAddons/ScratchAddons/labels/status%3A%20invalid from `#b60205` to `#cfd3d7` or a less distracting shade of red? Invalid issues aren't really supposed to be eye candy.

**Reply by BroJac5246** [reply]
> And maybe we should change the color of https://github.com/ScratchAddons/ScratchAddons/labels/status%3A%20invalid from `#b60205` to `#cfd3d7` or a less distracting shade of red? Invalid issues aren't really supposed to be eye candy.

Maybe it should be a less distracting shade, but red definitely feels like the right color.

**Comment by DNin01** [comment]
What do you think about a DevEx (developer experience) label? Would it be useful? We have https://github.com/ScratchAddons/ScratchAddons/labels/scope%3A%20documentation and https://github.com/ScratchAddons/ScratchAddons/labels/scope%3A%20meta but sometimes DevEx-related issues don't get those labels.

**Reply by WorldLanguages** [reply]
sure

**Reply by DNin01** [reply]
Or maybe it can be a generalized "scope: development" label for everything related to how we develop, likely mostly comprising of DevEx stuff.

**Reply by DNin01** [reply]
https://github.com/ScratchAddons/ScratchAddons/labels/scope%3A%20development

**Comment by DNin01** [comment]
Just created a https://github.com/ScratchAddons/ScratchAddons/labels/platform%3A%20mobile label.

**Comment by DNin01** [comment]
I have changed the color of the https://github.com/ScratchAddons/ScratchAddons/labels/scope%3A%20from%20turbowarp label from `#c5def5` to `#ff4c4c`.

**Reply by Samq64** [reply]
Now for chrome and Firefox

**Reply by DNin01** [reply]
> Now for chrome and Firefox

The reason I chose this particular color for this label is because it's actually TurboWarp's primary brand color.

**Reply by Samq64** [reply]
Yeah I know I was joking about the other labels

**Reply by mxmou** [reply]
Firefox could have a gradient if GitHub supported that.
![image](https://github.com/ScratchAddons/ScratchAddons/assets/51849865/994990db-a843-411c-ab3e-71f680d23885)

**Reply by WorldLanguages** [reply]
lol

**Reply by WorldLanguages** [reply]
Worth considering: a label specifically for CSS/layout issues (we have `scope: design` but we rarely use it for CSS bugs in the Scratch editor addons and so on)

**Reply by DNin01** [reply]
> Worth considering: a label specifically for CSS/layout issues (we have `scope: design` but we rarely use it for CSS bugs in the Scratch editor addons and so on)

Could be called "scope: style" or "scope: styling"?

**Comment by DNin01** [comment]
Clarified the Manifest V3 labels:
```diff
- mv3: breaking changes (Manifest v3 breaking changes)
- mv3: transition (Changes we'd need to make to keep us with manifest v3)
+ mv3: breaking changes (Forwards-incompatible changes that must be made to upgrade to Manifest V3)
+ mv3: transition (Forwards-compatible changes we can make anytime before the upgrade to Manifest V3)
```

**Reply by WorldLanguages** [reply]
Those labels are so old they kind of lost their meaning, but the change looks fine
A few years back I thought Google would be better at this

**Comment by Samq64** [comment]
I added https://github.com/ScratchAddons/ScratchAddons/labels/scope%3A%20addon%20compatiblity for compatibility issues between multiple addons. Feel free to adjust it.

**Comment by DNin01** [comment]
Should we change the color of https://github.com/ScratchAddons/ScratchAddons/labels/scope%3A%20addon%2Ejson to something like mint green? Being pink currently, when trying to triage quickly, it can be easily mistaken for the https://github.com/ScratchAddons/ScratchAddons/labels/scope%3A%20addon label.

**Reply by WorldLanguages** [reply]
Sure, change it

**Reply by DNin01** [reply]
Done.

**Comment by DNin01** [comment]
Would it be cool to have labels for "small" or "extra small" pull requests that can be reviewed very quickly?

**Reply by Hans5958** [reply]
Why not https://github.com/ScratchAddons/ScratchAddons/labels/priority%3A%204?

**Reply by BroJac5246** [reply]
> Why not https://github.com/ScratchAddons/ScratchAddons/labels/priority%3A%204?

Because that would say they're low priority and that speed therefore isn't important.

**Reply by DNin01** [reply]
> > Why not https://github.com/ScratchAddons/ScratchAddons/labels/priority%3A%204?
> 
> Because that would say they're low priority and that speed therefore isn't important.

An issue can be low-priority and still very simple to implement and review. Just because it's a quick fix doesn't mean it should override important updates or maintenance.

And regardless of priority, many factors such as cost (time, resources, etc.) and what is planned affect how long something takes to do. Small to-dos get done faster not because they are high-priority, but because they don't require many resources.

Still, I think having size tags would be useful information for anyone trying to find something that they can take a look at in the time that they have. Quick-to-review PRs may be tagged with low-priority, but that doesn't mean that low-priority PRs are quick to review.

**Reply by WorldLanguages** [reply]
> Would it be cool to have labels for "small" or "extra small" pull requests that can be reviewed very quickly?

Sure

**Reply by DNin01** [reply]
I've started with https://github.com/ScratchAddons/ScratchAddons/labels/size%3A%20extra%20small and https://github.com/ScratchAddons/ScratchAddons/labels/size%3A%20small. These labels aren't necessary for every issue.

**Reply by Samq64** [reply]
Would complexity labels make more sense?

**Reply by Hans5958** [reply]
Size and complexity are two different metrics, by the way.
